Lens fails to install bundled extensions on Linux/AppImage

Because "npm install" does not support read-only filesystems.

```
npm ERR! code EROFS
npm ERR! syscall access
npm ERR! path /tmp/.mount_Lens-4n78npS/resources/extensions/metrics-cluster-feature
npm ERR! errno -30
npm ERR! rofs EROFS: read-only file system, access '/tmp/.mount_Lens-4n78npS/resources/extensions/metrics-cluster-feature'
npm ERR! rofs Often virtualized file systems, or other file systems
npm ERR! rofs that don't support symlinks, give this error.
```

--- a/src/extensions/extension-manager.ts
+++ b/src/extensions/extension-manager.ts
@@ -25,8 +25,16 @@ export class ExtensionManager {
     return extensionPackagesRoot()
   }
 
+  get inTreeTargetPath() {
+    return path.join(this.extensionPackagesRoot, "extensions")
+  }
+
   get inTreeFolderPath(): string {
     return path.resolve(__static, "../extensions");
+  }
+
+  get nodeModulesPath(): string {
+    return path.join(this.extensionPackagesRoot, "node_modules")
   }
 
   get localFolderPath(): string {
@@ -39,7 +47,12 @@ export class ExtensionManager {
 
   async load() {
     logger.info("[EXTENSION-MANAGER] loading extensions from " + this.extensionPackagesRoot)
-    await fs.ensureDir(path.join(this.extensionPackagesRoot, "node_modules"))
+    if (this.inTreeFolderPath !== this.inTreeTargetPath) {
+      await fs.remove(this.inTreeTargetPath)
+      await fs.ensureDir(this.inTreeTargetPath)
+      await fs.copy(this.inTreeFolderPath, this.inTreeTargetPath)
+    }
+    await fs.ensureDir(this.nodeModulesPath)
     await fs.ensureDir(this.localFolderPath)
     return await this.loadExtensions();
   }
@@ -55,7 +68,7 @@ export class ExtensionManager {
         id: manifestJson.name,
         version: manifestJson.version,
         name: manifestJson.name,
-        manifestPath: path.join(this.extensionPackagesRoot, "node_modules", manifestJson.name, "package.json"),
+        manifestPath: path.join(this.nodeModulesPath, manifestJson.name, "package.json"),
         manifest: manifestJson
       }
     } catch (err) {
@@ -87,7 +100,7 @@ export class ExtensionManager {
 
   async loadBundledExtensions() {
     const extensions: InstalledExtension[] = []
-    const folderPath = this.inTreeFolderPath
+    const folderPath = this.inTreeTargetPath
     const bundledExtensions = getBundledExtensions()
     const paths = await fs.readdir(folderPath);
     for (const fileName of paths) {

--- a/src/extensions/extension-manager.ts
+++ b/src/extensions/extension-manager.ts
@@ -48,6 +48,7 @@ export class ExtensionManager {
   async load() {
     logger.info("[EXTENSION-MANAGER] loading extensions from " + this.extensionPackagesRoot)
     if (this.inTreeFolderPath !== this.inTreeTargetPath) {
+      // we need to copy in-tree extensions so that we can symlink them properly on "npm install"
       await fs.remove(this.inTreeTargetPath)
       await fs.ensureDir(this.inTreeTargetPath)
       await fs.copy(this.inTreeFolderPath, this.inTreeTargetPath)

